gapic/schema/api.py

```diff
@@ -560,6 +561,133 @@ def mixin_http_options(self):
             res[s] = [rule for rule in opt_gen if rule]
         return res
 
+    def get_auto_populated_fields_candidates(
+        self,
+        message: MethodDescriptorProto,
+    ) -> Sequence[str]:
+        """Return fields which meet the criteria to be automatically populated for a given message.
+        For a field to be automatically populated, all the below configurations must
+        be true according to https://google.aip.dev/client-libraries/4235:
+
+        - The field must be of type string
+        - The field must be at the top-level of the request message
+        - The field must not be annotated with google.api.field_behavior = REQUIRED.
+        - The field must be annotated with google.api.field_info.format = UUID4.
```

Comment:
Add
```
        The field presence requirements in that AIP are checked at run time.
```

Right?

Reply:
Fixed in https://github.com/googleapis/gapic-generator-python/commit/12f4333d8881e0fd284199ae375b599185ce6ddd